### PR TITLE
only one aggregatescore on heatmap metric selector

### PR DESCRIPTION
**Pull request by cdebourcy**
Show only one aggregatescore, not NT.aggregatescore and NR.aggregatescore (they are the same) -- JIRA-337. 
Tiago's & Jonathan's comment that I should define a display name led to the realization that such names are defined 5 times over in different parts of the app. This is a first attempt at standardization so we only need to change code in 1 place and all metric names are consistent.

### CODE REVIEW

**Comment by jshoe** [comment]
should NT.aggregatescore be renamed?

File: app/controllers/samples_controller.rb [hunk 1]
```diff
@@ -228,7 +228,6 @@ def heatmap
         "NT.r",
```
Comment (tfrcarvalho) [inline app/controllers/samples_controller.rb]: We should remove the NT too for the aggregate score.
The plan here is to send a full object: `[{name: ..., value: ...}]` so that metrics show up properly and values are still understanded by the server.

File: app/assets/src/components/utils/Metrics.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+let metric_names = {
```
Comment (tfrcarvalho) [inline app/assets/src/components/utils/Metrics.js]: replace by `const MetricName`;

File: app/controllers/samples_controller.rb [hunk 3]
```diff
@@ -223,36 +223,18 @@ def heatmap
         Viruses: ["Phage"]
       },
       metrics: [
+        "NT.r", # first entry will be default selection
```
Comment (tfrcarvalho) [inline app/controllers/samples_controller.rb]: Better do default selection in the client. Just select the value instead of the first one, so that the metrics are ordered.

Comment (cdebourcy) [reply app/controllers/samples_controller.rb]: ok

File: app/assets/src/components/utils/Metrics.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+for (var m in metric_names) {
+  metric_names_augmented[m] = metric_names[m];
+  metric_names_augmented[m.replace(".", "_")] = metric_names[m];
+  metric_names_augmented[m.replace(".", "_").toLowerCase()] = metric_names[m];
+}
+metric_names_augmented["aggregatescore"] =
+  metric_names_augmented["NT.aggregatescore"];
+export const METRIC_NAMES = metric_names_augmented;
+
+export const THRESHOLD_METRICS = [
```
Comment (tfrcarvalho) [inline app/assets/src/components/utils/Metrics.js]: These are the name that will show up? 
Should we use proper spacing and words, i.e. for instance `NT Percent Identity`

File: app/assets/src/components/views/TaxonTreeVis.jsx [hunk 5]
```diff
@@ -30,15 +31,33 @@ class TaxonTreeVis extends React.Component {
 
     this.metrics = {
       aggregatescore: {
-        label: "Aggregate Score",
+        label: METRIC_NAMES["aggregatescore"],
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/TaxonTreeVis.jsx]: No need to add label here. Remove the labels here and call `METRIC_NAMES[metric]` below instead.

File: app/controllers/samples_controller.rb [hunk 6]
```diff
@@ -223,36 +223,18 @@ def heatmap
...
         "NT.rpm",
-        "NT.r",
         "NT.zscore",
         "NT.maxzscore",
-        "NR.aggregatescore",
         "NR.rpm",
         "NR.r",
-        "NR.zscore",
-        "NR.maxzscore"
+        "NR.zscore"
```
Comment (tfrcarvalho) [inline app/controllers/samples_controller.rb]: Why did you remove NR.maxzscore?

File: app/controllers/samples_controller.rb [hunk 7]
```diff
@@ -223,36 +223,18 @@ def heatmap
         Viruses: ["Phage"]
       },
       metrics: [
+        "NT.r",
```
Comment (tfrcarvalho) [inline app/controllers/samples_controller.rb]: You already have default value above. You can reorder these.

File: app/assets/src/components/utils/Metrics.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
+const metric_names = {
```
Comment (tfrcarvalho) [inline app/assets/src/components/utils/Metrics.js]: Naming to camel case.

File: app/assets/src/components/utils/Metrics.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  "NT.maxzscore": "Max z-score",
+  "NT.percentidentity": "NT %id",
+  "NT.neglogevalue": "NT log(1/e)",
+  "NR.rpm": "NR rPM",
+  "NR.r": "NR r (reads)",
+  "NR.zscore": "NR z-score",
+  "NR.percentidentity": "NR %id",
+  "NR.neglogevalue": "NR log(1/e)"
+};
+const metric_names_augmented = {};
```
Comment (tfrcarvalho) [inline app/assets/src/components/utils/Metrics.js]: Naming to camelCase.

File: app/controllers/samples_controller.rb (hunk 6 above)
Comment (cdebourcy) [reply app/controllers/samples_controller.rb]: because it is the same as NT.maxzscore. `maxzscore == max(NT.zscore, NR.zscore)`

File: app/assets/src/components/views/TaxonTreeVis.jsx (hunk 5 above)
Comment (cdebourcy) [reply app/assets/src/components/views/TaxonTreeVis.jsx]: thanks

File: app/controllers/samples_controller.rb (hunk 7 above)
Comment (cdebourcy) [reply app/controllers/samples_controller.rb]: ok

File: app/assets/src/components/utils/Metrics.js (hunk 8 above)
Comment (cdebourcy) [reply app/assets/src/components/utils/Metrics.js]: ok

File: app/assets/src/components/utils/Metrics.js (hunk 9 above)
Comment (cdebourcy) [reply app/assets/src/components/utils/Metrics.js]: ok

File: app/controllers/samples_controller.rb (hunk 1 above)
Comment (cdebourcy) [reply app/controllers/samples_controller.rb]: fixed

File: app/controllers/samples_controller.rb (hunk 6 above)
Comment (tfrcarvalho) [reply app/controllers/samples_controller.rb]: I see. It confused me because here still has the NT prefix. Thanks

File: app/assets/src/components/views/TaxonTreeVis.jsx [hunk 10]
```diff
@@ -30,15 +31,26 @@ class TaxonTreeVis extends React.Component {
...
-        label: "Aggregate Score",
         agg: arr => Math.max(...arr)
       },
-      nt_r: { label: "NT r", agg: arr => arr.reduce((a, b) => a + b, 0) },
-      nt_rpm: { label: "NT rpm", agg: arr => arr.reduce((a, b) => a + b, 0) },
-      nt_zscore: { label: "NT Z-Score", agg: arr => Math.max(...arr) },
-      nr_r: { label: "NR r", agg: arr => arr.reduce((a, b) => a + b, 0) },
-      nr_rpm: { label: "NR rpm", agg: arr => arr.reduce((a, b) => a + b, 0) },
-      nr_zscore: { label: "NR Z-Score", agg: arr => Math.max(...arr) }
+      nt_r: {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/TaxonTreeVis.jsx]: Sorry for another ask, but we do no need the objects now. Can be just a mapping of metrics to aggregators.

Comment (cdebourcy) [reply app/assets/src/components/views/TaxonTreeVis.jsx]: What do you mean by objects?

Comment (cdebourcy) [reply app/assets/src/components/views/TaxonTreeVis.jsx]: oh i see

File: app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx [hunk 11]
```diff
@@ -3,14 +3,18 @@ import { forbidExtraProps } from "airbnb-prop-types";
 import PrimaryButton from "../buttons/PrimaryButton";
 import PropTypes from "prop-types";
 import RemoveIcon from "../../icons/RemoveIcon";
+import {
```
Comment (tfrcarvalho) [inline app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx]: Can we keep this drop down generic, please? 
Use the props as before and pass the values from outside.

Comment (cdebourcy) [reply app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx]: Well, it is called "ThresholdFilterDropdown". Why is it inappropriate to use a constant called THRESHOLD_METRICS if this is the level of abstraction we're operating at?

Comment (cdebourcy) [reply app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx]: I added this constant here because the props in all places where it was used for duplicated. IMO, we can reevaluate if we ever start using a threshold filter for metrics that are not related to the current ones. Right now, this use case is not even on the horizon, so I feel using props for data that never changes is misleading

Comment (tfrcarvalho) [reply app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx]: Because you cannot ever customize the metrics you want to use it with for different pages.
ThresholdFilterDropdown only means that you are applying a group of logical comparisons to a group of metrics.
I made it generic on purpose for easy reusability. 

What you are doing here is simply passing props, thorugh constants.

If you really want to make a default one, use default props to set defaults.
Also, please consolidate the two imports by defining something that equates to `MetricTextAndValue(THRESHOLD_METRICS)` in the `Metrics.js`.

Comment (cdebourcy) [reply app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx]: ok
